Holtron.Net/Network/Encryption/NetEncryptionAESGCM.cs

```diff
@@ -0,0 +1,82 @@
+using System.Security.Cryptography;
+
+namespace Holtron.Net.Network.Encryption
+{
+    public class NetEncryptionAESGCM : INetEncryption
+    {
+        // This implementation is from
+        // https://stackoverflow.com/questions/60889345/using-the-aesgcm-class
+        private readonly AesGcm aesGcm;
+
+        public NetEncryptionAESGCM(byte[] key)
+        {
+            var derivedKey = new Rfc2898DeriveBytes(key, new byte[8], 1000).GetBytes(16);
```

Comment:
- `Rfc2898DeriveBytes` is disposable so its usage should probably be accompanied by a `using` unless you want to keep the instance of `Rfc2898DerivedBytes` around.
- Was using `{ 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 }` as a salt intentional?
- I would move the number of rounds for PBKDF2 (`Rfc2898DeriveBytes`) into a class-level private `const`. You may also want to consider using more than 1000 rounds. It looks like OWASP recommends a significantly larger number of rounds: [Password Storage Cheat Sheet (PBKDF2)](https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#pbkdf2).